The token request should not include the scope parameter

anthony-vito opened the pull request:
The scope parameter is not indicated in RFC7636 for inclusion in the token request. It should be removed as it causes issues with strict IDP implementations.

## What does this pull request change?

Removes "scope" parameter from token request

## Why is this pull request needed?

Including "scope" is a specification violation

## Issues related to this change

https://github.com/soofstad/react-oauth2-pkce/issues/129

soofstad commented:
Good stuff! :+1: 
Just want to add that "Including "scope" is a specification violation" is not really the reason to cut this. The package quite often goes outside the spec to try and make the flow work with "non-rfc-compliant-IDP"'s.

But now that we have one example of an IDP that will fail with "scope" in, and non that require "scope" to be there. I'd say it's safe to remove it.

soofstad commented on src/authentication.ts:
```diff
@@ -81,7 +81,6 @@ export const fetchTokens = (config: TInternalConfig): Promise<TTokenResponse> =>
   const tokenRequest: TTokenRequestWithCodeAndVerifier = {
```
Should update the type here as well

sebastianvitterso replied:
https://github.com/soofstad/react-oauth2-pkce/blob/a23e7081a7ce85c1b5a16cf310a2bb0e6fd4a0b6/src/Types.ts#L3-L13

anthony-vito replied:
Yes. Good catch. I'll update it.

soofstad commented:
I saw the tests had failed to run. So I fixed that and rebased your branch.  You can now see that they fail.
Note that "scope" should be allowed in the refresh request.

anthony-vito commented:
I moved the optional scope field to the token refresh request type to correct that. I also updated the test data since the scope parameter is no longer sent in the token request.

soofstad reviewed (approved):
Looks good! Thanks for the contribution
